### PR TITLE
Rename ModelState (the type) -> ModelStateEntry

### LINKED ISSUE #3326: Rename ModelState (the type) -> ModelStateEntry
See: https://github.com/aspnet/Mvc/issues/3315

`ModelState` (the type) rarely appears in user-code. Users will generally interact with the MSD directly using `AddModelError` or something similar. Seeing `ModelState` (the type) in user-code is usually a mistake. We should rename it raise our sanity level and prevent mistake-making by future generations of humans and cyborgs alike.

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes #3326

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @rynowak

File: src/Microsoft.AspNet.Mvc.ViewFeatures/Internal/ValidationHelpers.cs [hunk 1]
```diff
@@ -27,13 +27,13 @@ public static string GetUserErrorMessageOrDefault(ModelError modelError, ModelSt
...
         // Returns non-null list of model states, which caller will render in order provided.
-        public static IEnumerable<ModelState> GetModelStateList(
+        public static IEnumerable<ModelStateEntry> GetModelStateList(
             ViewDataDictionary viewData,
             bool excludePropertyErrors)
         {
             if (excludePropertyErrors)
             {
-                ModelState ms;
+                ModelStateEntry ms;
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.ViewFeatures/Internal/ValidationHelpers.cs]: entry

File: src/Microsoft.AspNet.Mvc.ViewFeatures/ViewFeatures/DefaultHtmlGenerator.cs [hunk 2]
```diff
@@ -546,7 +546,7 @@ public virtual IHtmlContent GenerateAntiforgery(ViewContext viewContext)
             }
 
             // If there are any errors for a named field, we add the css attribute.
-            ModelState modelState;
+            ModelStateEntry modelState;
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.ViewFeatures/ViewFeatures/DefaultHtmlGenerator.cs]: Should we rename all of these to `entry`? Your call if you think it's clearer

File: test/Microsoft.AspNet.Mvc.Abstractions.Test/ModelBinding/ModelStateDictionaryTest.cs [hunk 3]
```diff
@@ -408,8 +408,8 @@ public void IsValidPropertyReturnsFalse_IfSomeFieldsAreNotValidated()
         public void MergeCopiesDictionaryEntries()
         {
             // Arrange
-            var fooDict = new ModelStateDictionary() { { "foo", new ModelState() } };
-            var barDict = new ModelStateDictionary() { { "bar", new ModelState() } };
+            var fooDict = new ModelStateDictionary() { { "foo", new ModelStateEntry() } };
+            var barDict = new ModelStateDictionary() { { "bar", new ModelStateEntry() } };
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.Abstractions.Test/ModelBinding/ModelStateDictionaryTest.cs]: ugh these names are bad

**Comment by rynowak** [comment]
:shipit:

File: src/Microsoft.AspNet.Mvc.ViewFeatures/ViewFeatures/DefaultHtmlGenerator.cs (hunk 2 above)
Comment (pranavkm) [inline src/Microsoft.AspNet.Mvc.ViewFeatures/ViewFeatures/DefaultHtmlGenerator.cs]: Sure

**Pull request closed without merging**